### ISSUE TITLE
inaccessibility of many/all buttons with screen readers

### ISSUE THREAD

**Issue by lilmike** [thelounge/lounge#296]
Hi,
I am blind, so use a screen reader to access my browser, including my self-hosted lounge server. However, there are many buttons that are not labelled, so my screen reader simply reports them as "button," which is pretty useless to me :P. I just wanted to bring this to your attention so if possible it could be fixed, as this seems like a nice server for webirc.
Thanks,
-Michael.

**Comment by astorije** [comment]
Hi @lilmike,
Thanks for this report!
I'll try to take a look at those unlabeled buttons as soon as possible.
Out of curiosity, what screen reader are you using? I'll need to test the result with a screen reader and I don't have one at the moment.

**Comment by xPaw** [comment]
Just wondering, is it just the buttons you're having trouble with, or are there other places that could be improved?

**Comment by lilmike** [comment]
Hi,

I'm using Jaws for windows, but similar screen readers exist (voice over 
-- built into mac OSX, NVDA (windows, free), orca on linux).

-Michael.

**Comment by lilmike** [comment]
So far that's all I'm having trouble with, but I haven't been able to 
test much else than sending a message as I don't wish to break anything 
by pressing unknown buttons lol.

-Michael.

**Comment by xPaw** [comment]
We should be setting aria labels on text inputs as well, like the message input itself, search input. Also labels in connect/login window need to be correctly linked together.

**Comment by xPaw** [comment]
Just noticed we already have #188 open

**Comment by lilmike** [comment]
Just FYI, that's not the same thing as I'm talking about, though can 
definitely help with accessibility.

-Michael.

**Comment by xPaw** [comment]
The connect window should make more sense now, I've fixed labels there: e1a8baa49d7784c4a45163ec04e623d6063f70e5

Buttons are still left.

**Comment by astorije** [comment]
@lilmike, @xPaw, I'll take care of the buttons over the weekend.

**Comment by astorije** [comment]
`aria-labels` added to the icon (text-less) buttons in #303.
